### PR TITLE
Syntax lookup: add ref value assignment documentation

### PR DESCRIPTION
**Pull request by illusionalsagacity**
for #162

### CODE REVIEW

File: misc_docs/syntax/operator_ref_assignment.mdx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+id: "ref-assignment"
+keywords: ["ref", "assign"]
+name: ":="
+summary: "This is the `ref assignment` operator."
+category: "operators"
+---
+
+This operator is syntactic sugar for assigning to the `contents` value of a `ref`
+
```
Comment (bobzhang) [inline misc_docs/syntax/operator_ref_assignment.mdx:10]: actually no sugar provided for `ref`, both `ref` and `:=` are functions in pervasives module

Comment (illusionalsagacity) [reply misc_docs/syntax/operator_ref_assignment.mdx:10]: Gotcha, I was essentially quoting from pages/docs/manual/latest/mutation.mdx:

```
Assign a new value to `myValue` like so:

<CodeTab labels={["ReScript", "JS Output"]}>

\```res example
myValue.contents = 6
\```
\```js
myValue.contents = 6;
\```

</CodeTab>

We provide a syntax sugar for this:

<CodeTab labels={["ReScript", "JS Output"]}>

\```res example
myValue := 6
\```
\```js
myValue.contents = 6;
\```

</CodeTab>
```

Should I re-work this language as well?

File: misc_docs/syntax/operator_ref_assignment.mdx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
+id: "ref-assignment"
+keywords: ["ref", "assign"]
+name: ":="
+summary: "This is the `ref assignment` operator."
+category: "operators"
+---
+
+This operator is syntactic sugar for assigning to the `contents` value of a `ref`
```
Comment (ryyppy) [inline misc_docs/syntax/operator_ref_assignment.mdx:9]: Let's keep it simple with the wording, no need to talk about the technical details if this is syntax sugar / a function. To be exact, `:=` is a whitelisted operator within the syntax, so it's kinda both, a function defined in `Pervasives`, and "special sugar"  in the parser.

```suggestion
This operator assigns a new value to a `ref`.
```

File: misc_docs/syntax/operator_ref_assignment.mdx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+```
+
+```js
+var total = {
+  contents: 0
+};
+
+total.contents = 1;
+```
+
```
Comment (ryyppy) [inline misc_docs/syntax/operator_ref_assignment.mdx:25]:
~~~suggestion
A `ref` is a builtin record type with a `mutable contents` field. The `:=` operator is just a shorthand version for the following code:

```res
let total = ref(0)
total.contents = 0
```
~~~

File: misc_docs/syntax/operator_ref_assignment.mdx (hunk 1 above)
Comment (ryyppy) [reply misc_docs/syntax/operator_ref_assignment.mdx:10]: I added some comments. I think it's not important how this is implemented, we should just describe what it does and how it is useful.

File: misc_docs/syntax/operator_ref_assignment.mdx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+---
+id: "ref-assignment"
```
Comment (ryyppy) [inline misc_docs/syntax/operator_ref_assignment.mdx:2]:
```suggestion
id: "ref-value-assignment"
```

File: misc_docs/syntax/operator_ref_assignment.mdx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+keywords: ["ref", "assign"]
+name: ":="
+summary: "This is the `ref assignment` operator."
```
Comment (ryyppy) [inline misc_docs/syntax/operator_ref_assignment.mdx:5]:
```suggestion
summary: "This is the `ref value assignment` operator."
```

**Comment by ryyppy** [comment]
@illusionalsagacity looking good, thanks!

**Pull request merged**